libs/parser/src/Parser.php

```diff
@@ -236,9 +253,37 @@ private function parseOrFail(Context $context): iterable
             return $result;
         }
 
-        $token = $context->lastOrdinalToken ?? $context->buffer->current();
+        $token = clone($context->lastOrdinalToken ?? $context->buffer->current());
+
+        if ($this->possibleTokensSearching) {
+            $this->findPossibleTokensForUnexpected($context);
+        }
+
+        throw UnexpectedTokenWithHintsException::fromToken($context->getSource(), $token, null, $this->possibleTokens);
+    }
+
+    /**
+     * @param Context $context
+     * @return void
+     */
+    private function findPossibleTokensForUnexpected(Context $context): void
+    {
+        $problemTokenOffset = $context->lastOrdinalToken->getOffset();
+        $problemTokenKey = 0;
+        while ($context->buffer->get($problemTokenKey)->getOffset() != $problemTokenOffset) {
```

Comment:
Although in this case int + int is compared, but using non-strict `!=` comparisons does not seem to me a good idea. But you can leave it as it is now. Up to you =)

Reply:
Correct, it is mistake. Fix it.